local-dev/README.md

```diff
@@ -0,0 +1,310 @@
+
+Also note that instructions in this doc presume that relation engine indexing is disabled.
+
+#### `SKIP_FEATURES`
+
+boolean string value (yes, no, y, n, true, false, t, f, 1, 0) defaults to `yes`.
+
+If true, the default, Genome features will not be indexed. Indexing genomes can dramatically increase the time it takes to index workspaces containing Genomes. Unless you are testing genome feature indexing, it is best to disable this feature.
+
+#### `MAX_OBJECT_SIZE`
+
+an integer string value, defaults to "100000000" (100 million)
+
+Only objects at or below this value will be indexed. This is handy for local development which is usually operating in a resource constrained environment (your laptop!). Disabling this (by setting it to an empty string) is not advisable, as a large object can crash the indexer. This can interrupt and otherwise easy indexing process.
+
+#### `HOST`
+
+a hostname, defaults to `ci.kbase.us`
+
+The host for forming urls to KBase services may be specified, and defaults to the CI host.
+
+Note that the `HOST_IP` needs to be specified if the `HOST` is.
+
+#### `HOST_IP`
+
+an ip address, defaults to `128.3.56.133`.
+
+When you are using a local proxy to kbase, the indexer running inside the docker container will consult your host for hostname lookup. E.g. if you are proxying `ci.kbase.us` in order to route such calls locally, a docker container which needs to contact the "real" ci.kbase.us will instead talk to the proxy. Ordinarily this would be fine -- the proxy would route the request to the real ci.kbase.us. But in local development with the kbase-ui proxy, the dev ssl cert will not be recognized by the indexer, and will throw an error.
+
+Future work make allow us to overcome this, e.g. by installing the local dev cert for *.kbase.us in the indexer container.
+
+In any case, the `HOST_IP` setting, in combination with the `HOST`, is used to set up a special DNS route for the container, via the `-add-host` option.
+
+## Indexing Test Data
+
+Local indexing is conducted against an active KBase deployment which supports at least the following services:
+
+- workspace
+- auth
+- user profile
+
+and in addition, indexers may rely upon other core or dynamic services.
+
+The `local-dev` directory contains a number of scripts with which you can index entire classes of workspaces. The scope of these scripts is at the workspace level, so all objects within a workspace will be indexed.
+
+Well, that is a lie. There are object-level filters for:
+
+- object size (`MAX_OBJECT_SIZE`)
+- blacklisted types (defined the indexer spec)
+
+The scripts are easy enough to understand that if you have a special case, it should be feasible to create a custom indexing script. If it is generally useful, check it in!
+
+### Using the scripts
+
+The scripts are, at present, implemented in Ruby. All you need is a relatively recent version of Ruby 2.X. No external dependencies are required.
```

Comment:
See my notes below. I would prefer not to add ruby to the mix here.  Can't these be done in bash or python?